flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/SlotContext.java

```diff
@@ -58,4 +36,11 @@
 	 * @return The gateway that can be used to send messages to the TaskManager.
 	 */
 	TaskManagerGateway getTaskManagerGateway();
+
+	/**
+	 * Returns the resource profile of the slot.
+	 *
+	 * @return the resource profile of the slot.
+	 */
+	ResourceProfile getResourceProfile();
```

Comment:
Do we need to add this method in this PR?

Reply:
Agreed, we can remove it completely for this PR.